src/functional/iterator/embedded.py

```diff
@@ -317,6 +318,16 @@ def less(first, second):
     return first < second
 
 
+for math_builtin_name in builtins.MATH_BUILTINS:
+    decorator = getattr(builtins, math_builtin_name).register(EMBEDDED)
+    if math_builtin_name == "gamma":
+        # numpy has no gamma function
+        impl = np.vectorize(math.gamma)
+    else:
+        impl = getattr(np, math_builtin_name)
+    globals()[math_builtin_name] = decorator(impl)
```

Comment:
```suggestion
    decorator(impl)
```
no need to add the function to the module

Reply:
All other builtins are accessible from the module so deviating from that for the math builtins doesn't feel right to me. If we don't expose them here, I would argue the decorator of the other builtins should not expose them either. I personally prefer exposing all of them, but making it a convention to not use them directly.